### DISCUSSION TITLE
Why does Vercel show fewer static HTML files than my local build?

### DISCUSSION THREAD

**Discussion by ben519** [vercel/next.js#50621] [Help] [answered]
### Summary

I'm trying to wrap my head around pre-rendering and Static Site Generation. Here's what I see when I run `npm run build` with the `--debug` flag on my local machine.

<img width="1291" alt="local" src="https://github.com/vercel/next.js/assets/723099/53c84e15-1d91-4dab-8957-42ce45d1994e">

Notice it says "Generating static pages (5/5)". I believe these are the html pages shown on the left, namely

- `.next/server/app/user-server/1.html`
- `.next/server/app/user-server/2.html`
- `.next/server/app/index.html`
- `.next/server/pages/404.html`
- `.next/server/pages/500.html`

Now here's what I see after deploying the same code to Vercel.

<img width="1223" alt="vercel" src="https://github.com/vercel/next.js/assets/723099/4b924a27-2f60-4897-ae9f-0cc1e353ed91">

Notice it says there are two HTML files, namely

- `404`
- `500`

Why is my local build different than Vercel? A general explanation of what I'm looking at would be awesome. Thanks!

**Comment by devjiwonchoi** [comment]
In the photo you've provided, the `next build` contains a flag `--debug`. 
Assuming that you are migrating from the `Page Router` to `App Router`, the previous `Page Router` had [few options](https://nextjs.org/docs/pages/api-reference/next-cli#build) for `next build`.

When using the `--debug` flag, it is natural that the build creates more pages than the actual production 
since it allows for generating additional build outputs.

`next build --debug`
> With this flag enabled additional build output like rewrites, redirects, and headers will be shown.

Please double check your `package.json` whether your `build` script is set to `next build --debug`.

If the script does not contain the `--debug` flag, please try to run 

```npm run next build```

and see if it runs on the debug mode by default.

Another piece of advice is to update Next.js to the latest version by 
```
npm install next@latest react@latest react-dom@latest
```
or fully migrate from the `Page Router` to the `App Router` which can surely resolve your issue.

**Reply by ben519** [reply]
Thanks for the reply.

> or fully migrate from the `Page Router` to the `App Router` ...

What gives you the impression I'm using the `Page Router`?

I've removed the `--debug` flag and there is still the same discrepancy between Vercel and my local build.

**Reply by devjiwonchoi** [reply]
This is my hypothesis:

Since the `App Router` has been `experimental` lately, 
the `Deployment Status` on Vercel might not be updated to catch files inside the `app` directory.

This is based on the [official docs](https://nextjs.org/docs/app/building-your-application/deploying#managed-nextjs-with-vercel) about `Managed Next.js with Vercel` since it hasn't been updated 
and is identical to the `Page Router` [docs](https://nextjs.org/docs/pages/building-your-application/deploying#managed-nextjs-with-vercel).

I'm sure it's nothing to do about `next`, but with `Vercel Deployment`.

**Comment by icyJoseph** [comment, marked as answer]
Yeah I'd guess that maybe that section of the `Deployment` is not yet finding the APP router .html pages, at least not on the Vercel builder version we have. 

I for once, would trust much more in the `Source`:

<img width="752" alt="Screenshot 2023-06-01 at 18 45 51" src="https://github.com/vercel/next.js/assets/21013447/11eb9864-2e39-48cf-a122-a172190a9860">

Might be worth opening an issue with the Vercel support team anyway. I think that's the proper channel.

**Reply by ben519** [reply]
Thanks, really appreciate your help!
